File: app_manager/appmanager/src/main/java/app/editors/manager/ui/dialogs/ActionBottomDialog.java
package app.editors.manager.ui.dialogs;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import app.editors.manager.R;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import lib.toolkit.base.ui.dialogs.base.BaseBottomDialog;

public class ActionBottomDialog extends BaseBottomDialog {

    public static final String TAG = ActionBottomDialog.class.getSimpleName();
    private static final String TAG_THIRD_PARTY = "TAG_THIRD_PARTY";
    private static final String TAG_DOCS = "TAG_DOCS";
    private static final String TAG_LOCAL = "TAG_LOCAL";

    public enum Buttons {
        NONE, SHEET, PRESENTATION, DOC, FOLDER, PHOTO, UPLOAD, STORAGE
    }

    public interface OnClickListener {
        void onActionButtonClick(Buttons buttons);
    }

    @BindView(R.id.list_explorer_action_sheet)
    protected LinearLayout mListActionSheet;
    @BindView(R.id.list_explorer_action_presentation)
    protected LinearLayout mListActionPresentation;
    @BindView(R.id.list_explorer_action_docs)
    protected LinearLayout mListActionDocs;
    @BindView(R.id.list_explorer_action_folder)
    protected LinearLayout mListActionFolder;
    @BindView(R.id.list_explorer_action_photo)
    protected LinearLayout mListActionPhoto;
    @BindView(R.id.list_explorer_action_upload)
    protected LinearLayout mListActionUpload;
    @BindView(R.id.view_line_separator_storage)
    protected View mViewLineSeparatorStorage;
    @BindView(R.id.list_explorer_action_storage)
    protected LinearLayout mListActionStorage;

    @Nullable
    private Unbinder mUnbinder;
    private OnClickListener mOnClickListener;
    private boolean mIsThirdParty = false;
    private boolean mIsDocs = true;
    private boolean mIsLocal = false;
    private boolean mIsWebDav = false;

    public static ActionBottomDialog newInstance() {
        return new ActionBottomDialog();
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final Dialog dialog = super.onCreateDialog(savedInstanceState);
        restoreValues(savedInstanceState);
        return dialog;
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putBoolean(TAG_THIRD_PARTY, mIsThirdParty);
        outState.putBoolean(TAG_DOCS, mIsDocs);
        outState.putBoolean(TAG_LOCAL, mIsLocal);
    }

    @SuppressLint("RestrictedApi")
    @Override
    public void setupDialog(@NonNull Dialog dialog, int style) {
        super.setupDialog(dialog, style);
        init(dialog);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (mUnbinder != null) {
            mUnbinder.unbind();
        }
        mOnClickListener = null;
    }

    @OnClick({R.id.list_explorer_action_sheet,
            R.id.list_explorer_action_presentation,
            R.id.list_explorer_action_docs,
            R.id.list_explorer_action_folder,
            R.id.list_explorer_action_photo,
            R.id.list_explorer_action_upload,
            R.id.list_explorer_action_storage})
    public void onButtonsClick(View view) {
        if (mOnClickListener != null) {
            switch (view.getId()) {
                case R.id.list_explorer_action_sheet:
                    mOnClickListener.onActionButtonClick(Buttons.SHEET);
                    break;
                case R.id.list_explorer_action_presentation:
                    mOnClickListener.onActionButtonClick(Buttons.PRESENTATION);
                    break;
                case R.id.list_explorer_action_docs:
                    mOnClickListener.onActionButtonClick(Buttons.DOC);
                    break;
                case R.id.list_explorer_action_folder:
                    mOnClickListener.onActionButtonClick(Buttons.FOLDER);
                    break;
                case R.id.list_explorer_action_photo:
                    mOnClickListener.onActionButtonClick(Buttons.PHOTO);
                    break;
                case R.id.list_explorer_action_upload:
                    mOnClickListener.onActionButtonClick(Buttons.UPLOAD);
                    break;
                case R.id.list_explorer_action_storage:
                    mOnClickListener.onActionButtonClick(Buttons.STORAGE);
                    break;
            }
            dismiss();
        }
    }

    private void restoreValues(final Bundle savedInstanceState) {
        if (savedInstanceState != null) {
            mIsThirdParty = savedInstanceState.getBoolean(TAG_THIRD_PARTY);
            mIsDocs = savedInstanceState.getBoolean(TAG_DOCS);
            mIsLocal = savedInstanceState.getBoolean(TAG_LOCAL);
        }
    }

    private void init(final Dialog dialog) {
        final View contentView = View.inflate(getContext(), R.layout.list_explorer_action_menu, null);
        mUnbinder = ButterKnife.bind(this, contentView);
        dialog.setContentView(contentView);
        dialog.setCanceledOnTouchOutside(true);
        setViewState();
    }

    private void setViewState() {
        if (!mIsThirdParty) {
            mViewLineSeparatorStorage.setVisibility(View.GONE);
            mListActionStorage.setVisibility(View.GONE);
        }

        if (!mIsDocs) {
            mListActionDocs.setVisibility(View.GONE);
            mListActionPresentation.setVisibility(View.GONE);
            mListActionSheet.setVisibility(View.GONE);
        }

        if (mIsLocal) {
            mListActionDocs.setVisibility(View.VISIBLE);
            mListActionPresentation.setVisibility(View.VISIBLE);
            mListActionSheet.setVisibility(View.VISIBLE);
            if (mIsWebDav) {
                mListActionUpload.setVisibility(View.VISIBLE);
            } else {
                mListActionUpload.setVisibility(View.GONE);
            }
        }
    }

    public void setOnClickListener(final OnClickListener onClickListener) {
        mOnClickListener = onClickListener;
    }

    public void setThirdParty(boolean thirdParty) {
        mIsThirdParty = thirdParty;
    }

    public void setDocs(boolean docs) {
        mIsDocs = docs;
    }

    public void setLocal(boolean isLocal) {
        mIsLocal = isLocal;
    }

    public void setWebDav(boolean mIsWebDav) {
        this.mIsWebDav = mIsWebDav;
    }
}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/dialogs/ActionBottomDialog.kt
package app.editors.manager.ui.dialogs

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import app.editors.manager.databinding.ListExplorerActionMenuBinding
import app.editors.manager.managers.utils.isVisible
import lib.toolkit.base.ui.dialogs.base.BaseBottomDialog

class ActionBottomDialog : BaseBottomDialog() {

    enum class Buttons {
        NONE, SHEET, PRESENTATION, DOC, FOLDER, PHOTO, UPLOAD, STORAGE
    }

    interface OnClickListener {
        fun onActionButtonClick(buttons: Buttons?)
        fun onActionDialogClose()
    }

    private var viewBinding: ListExplorerActionMenuBinding? = null
    var isThirdParty = false
    var isDocs = true
    var isLocal = false
    var isWebDav = false
    var onClickListener: OnClickListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        restoreValues(savedInstanceState)
        return super.onCreateDialog(savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putBoolean(TAG_THIRD_PARTY, isThirdParty)
        outState.putBoolean(TAG_DOCS, isDocs)
        outState.putBoolean(TAG_LOCAL, isLocal)
    }

    @SuppressLint("RestrictedApi")
    override fun setupDialog(dialog: Dialog, style: Int) {
        super.setupDialog(dialog, style)
        init(dialog)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        viewBinding = null
        onClickListener = null
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        onClickListener?.onActionDialogClose()
    }

    private fun initListeners() {
        viewBinding?.let {
            it.listExplorerActionDocs.setOnClickListener(Buttons.DOC)
            it.listExplorerActionFolder.setOnClickListener(Buttons.FOLDER)
            it.listExplorerActionPhoto.setOnClickListener(Buttons.PHOTO)
            it.listExplorerActionUpload.setOnClickListener(Buttons.UPLOAD)
            it.listExplorerActionStorage.setOnClickListener(Buttons.STORAGE)
            it.listExplorerActionSheet.setOnClickListener(Buttons.SHEET)
            it.listExplorerActionPresentation.setOnClickListener(Buttons.PRESENTATION)
        }
    }

    private fun View.setOnClickListener(button: Buttons) {
        this.setOnClickListener {
            onClickListener?.onActionButtonClick(button)
            dismiss()
        }
    }

    private fun restoreValues(savedInstanceState: Bundle?) {
        savedInstanceState?.let { state ->
            isThirdParty = state.getBoolean(TAG_THIRD_PARTY)
            isDocs = state.getBoolean(TAG_DOCS)
            isLocal = state.getBoolean(TAG_LOCAL)
        }
    }

    private fun init(dialog: Dialog) {
        viewBinding = ListExplorerActionMenuBinding.inflate(layoutInflater).apply {
            dialog.setContentView(root)
            dialog.setCanceledOnTouchOutside(true)
        }
        setViewState()
        initListeners()
    }

    private fun setViewState() {
        viewBinding?.let {
            it.viewLineSeparatorStorage.viewLineSeparator.isVisible = isThirdParty
            it.listExplorerActionStorage.isVisible = isThirdParty

            it.listExplorerActionDocs.isVisible = isDocs || isLocal
            it.listExplorerActionPresentation.isVisible = isDocs || isLocal
            it.listExplorerActionSheet.isVisible = isDocs || isLocal

            it.listExplorerActionUpload.isVisible = !isLocal || isWebDav
        }
    }

    companion object {

        val TAG: String = ActionBottomDialog::class.java.simpleName

        private const val TAG_THIRD_PARTY = "TAG_THIRD_PARTY"
        private const val TAG_DOCS = "TAG_DOCS"
        private const val TAG_LOCAL = "TAG_LOCAL"

        fun newInstance(): ActionBottomDialog {
            return ActionBottomDialog()
        }
    }
}
